python/ray/tests/test_failure.py

```diff
@@ -727,24 +728,44 @@ def test_warning_for_too_many_nested_tasks(shutdown_only):
     ray.init(num_cpus=num_cpus)
     p = init_error_pubsub()
 
+    remote_wait = Semaphore.remote(value=0)
+    nested_wait = Semaphore.remote(value=0)
+
+    ray.get([
+        remote_wait.locked.remote(),
+        nested_wait.locked.remote(),
+    ])
+
     @ray.remote
     def f():
         time.sleep(1000)
```

Comment:
i might be missing something here, but how does this line work?

Reply:
F is just long running (so that both h and g won't be terminated at all)

Reply:
i guess is there any concern about timing out with such a long sleep?

Reply:
This will be fine because the error check method is timed out within 5 seconds. This task will be just terminated after that.

Reply:
The point here is we won't ray.get on this line
```
[g.remote(remote_waits, nested_waits) for _ in range(num_root_tasks)]
```

Reply:
ahh k yeah we don't wait for the tasks to finish before the end of the test case. This was a weird way to write the test, but I guess that's not really the point of this PR.